parametric/test_span_sampling.py

```diff
@@ -442,3 +442,99 @@ def test_multi_rule_independent_rate_limiters_sss013(test_agent, test_library):
             pass
     span = find_span_in_traces(test_agent.wait_for_num_traces(1), Span(name="web.request"))
     assert span["metrics"].get(SINGLE_SPAN_SAMPLING_MAX_PER_SEC) == 1
+
+
+@pytest.mark.skip_library("golang", "Not implemented")
+@pytest.mark.skip_library("dotnet", "Not implemented")
+@pytest.mark.skip_library("nodejs", "Not implemented")
+@pytest.mark.skip_library("python", "Not implemented")
+@pytest.mark.parametrize(
+    "library_env",
+    [
+        {
+            "DD_SPAN_SAMPLING_RULES": json.dumps(
+                [{"service": "webserver", "name": "web.request", "sample_rate": 1.0, "max_per_second": 50}]
+            ),
+            "DD_TRACE_SAMPLING_RULES": json.dumps([{"sample_rate": 0}]),
+        }
+    ],
+)
+def test_root_span_selected_by_sss014(test_agent, test_library):
+    """Single spans selected by SSS must be kept and shouldn't affect child span sampling priority.
+    
+    We're essentially testing to make sure that the span sampling rule keeps selected spans regardless of the trace sampling decision
+    and doesn't affect child spans that are dropped by the tracer sampling mechanism.
+    """
+    with test_library:
+        with test_library.start_span(name="web.request", service="webserver") as parent_span:
+            with test_library.start_span(name="web.handler", service="webserver", parent_id=parent_span.span_id):
+                pass
+
+    traces = test_agent.wait_for_num_traces(1, clear=True)
+
+    root_span = find_span_in_traces(traces, Span(name="web.request", service="webserver"))
+    child_span = find_span_in_traces(traces, Span(name="web.handler", service="webserver"))
+
+    # root span should be kept by defined the SSS rules
+    assert root_span["metrics"].get(SAMPLING_PRIORITY_KEY) == USER_KEEP
```

Comment:
Is that true? The RFC doesn't say anything about changing the sampling priority when we don't have stats computations enabled.

Reply:
Agreed, the Python tracer passes everything except for these `sampling_priority_v1` assertions. Since stats computation is not turned on here, it sees like this is not the correct behavior: https://github.com/DataDog/architecture/blob/master/rfcs/apm/integrations/single-span-ingestion-control/README.md#when-tracers-dont-send-traces-to-the-agent

`test_keep_span_with_stats_computation_sss010` tests the behavior I would expect.

Reply:
The only sampling-priority-relevant check in these new tests ought to be that the root span priority is _not_ "keep" when span sampling is happening.

I think that this line can be removed.

Reply:
This line was already removed in the current commit of this PR, right?